apps/angular/bug-effect-signal/src/app/app.component.ts

```diff
@@ -28,6 +28,10 @@ export class AppComponent {
 
   constructor() {
     effect(() => {
+      console.log(this.name());
+      console.log(this.age());
+      console.log(this.address());
```

Comment:
This is a way to make the effect run, but you don't want to console log in production application.

Reply:
No, this isn't a real solution, but something I stumbled upon while working on the challenge.  And it shows the basic logic of what's happening.  

I actually think the best solution is to duplicate code and have multiple effects.

```
    effect(() => {
      if (this.name()) {
        alert('One checkbox is check');
      }
    });

  // etc
```
When you uncheck the box, the effect does not run again when you have multiple effects.

Reply:
yes I think that's the best solution. Easy to read, easy to maintain, easy to extend

Reply:
No, it is because of the short-circuiting that's why multiple effects are better for this challenge.  

Although `effect` can track multiple dependencies, it is really wonky (this challenge shows a edge case) and I think it is better to be certain that your effect will run when your signal changes.

Reply:
I agree with @jdegand. my solution will be to have one effect per signal, if there are independent of each other. We should always think of simplicity rather than cleverness, even if we need to repeat some part.